src/Factory/EventManagerFactory.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zend-eventmanager for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+namespace Zend\EventManager\Factory;
+
+use Zend\EventManager\EventManager;
+use Zend\EventManager\ListenerPluginManager;
+use Zend\ServiceManager\FactoryInterface;
+
+class EventManagerFactory implements FactoryInterface
+{
+    /**
+     * Override config key in an extending class to point to more specific key if needed
+     *
+     * @var string
+     */
+    protected $configKey = "event_manager";
+
+    /**
+     * {@inheritDoc}
+     */
+    public function __invoke(ServiceLocatorInterface $serviceLocator, $className, array $options = [])
```

Comment:
I have renamed the $className to $requestedName in latest proposal of SM, as it's not 100% for sure a class name :).